### DISCUSSION TITLE
Automation - retrieve value from previous row and add 1

### DISCUSSION THREAD

**Discussion by hendrikjap** [Budibase/budibase#4786] [Help] [answered]
I want to trigger automation = previous purchase id + 1

![image](https://user-images.githubusercontent.com/55825618/156694357-5871f65c-e738-4e24-b484-fbb8dffebf36.png)

How to accomplish this?

**Comment by mjashanks** [comment, marked as answer]
Hey @ely2727. 

You need to add a `Query Rows` step to retrieve the previous purchase. I'm assuming that the criteria for "Previous Purchase" is literally the most recently created purchase. Therefor, your `Query Rows` setup should be:

- No filter
- Sort by "Created Date", Descending .... (i.e. the most recent will appear first)
- Limit: 1 - i.e. it will only choose the first record

Then, in your `Create Row` step, you can get the `id` of the result from `Query Rows` by doing something like:

`{{ steps.1.rows.0.id }}`

`steps.1` - is the query rows step
`rows` - is the list of rows returned (a list of 1)
`.0` - chooses the first row
`.id` - choose a field called `id`

**Reply by hendrikjap** [reply]
It work @mjashanks ! Thanks !
